sys/include/net/ng_nettype.h

```diff
@@ -37,6 +37,11 @@ extern "C" {
  * @note    Expand at will.
  */
 typedef enum {
+    /**
+     * @brief   Protocol is as defined in @ref ng_netif_hdr_t. Not usable with
+     *          @ref net_ng_netreg
```

Comment:
Yes, the `ng_netreg` module. You can't register to type `NG_NETTYPE_NETIF` there.